linkhash.c

```diff
@@ -499,6 +499,8 @@ struct lh_table *lh_table_new(int size, lh_entry_free_fn *free_fn, lh_hash_fn *h
 	int i;
 	struct lh_table *t;
 
+	if (size < 1)
```

Comment:
I am not a fan of asserts since they are easily skipped from productive code. But as far as I understand lh_table_new is not exported for users of this library, so I am fine with it.